web/src/drafts.ts

```diff
@@ -274,40 +338,77 @@ export function update_draft(opts = {}) {
 
 export const DRAFT_LIFETIME = 30;
 
-export function remove_old_drafts() {
+export function remove_old_drafts(): void {
     const old_date = subDays(new Date(), DRAFT_LIFETIME).getTime();
     const drafts = draft_model.get();
     for (const [id, draft] of Object.entries(drafts)) {
-        if (draft.updatedAt < old_date) {
+        if (draft.updatedAt !== undefined && draft.updatedAt < old_date) {
             draft_model.deleteDraft(id);
         }
     }
 }
 
-export function format_draft(draft) {
+type FormattedDraft =
+    | {
+          is_stream: true;
+          draft_id: string;
+          stream_name: string;
+          recipient_bar_color: string;
+          stream_privacy_icon_color: string;
+          topic: string;
+          raw_content: string;
+          stream_id: number | undefined;
+          time_stamp: string;
+          invite_only: boolean;
+          is_web_public: boolean;
+      }
+    | {
+          is_stream: false;
+          draft_id: string;
+          recipients: string;
+          raw_content: string;
+          time_stamp: string;
+      };
+
+export function format_draft(draft: LocalStorageDraftWithId): FormattedDraft | undefined {
     const id = draft.id;
-    let formatted;
+    assert(draft.updatedAt !== undefined);
     const time = new Date(draft.updatedAt);
     let invite_only = false;
     let is_web_public = false;
     let time_stamp = timerender.render_now(time).time_str;
     if (time_stamp === $t({defaultMessage: "Today"})) {
         time_stamp = timerender.stringify_time(time);
     }
+
+    let markdown_data;
+    try {
+        markdown_data = markdown.get_parsed_data(draft.content);
+    } catch (error) {
+        // In the unlikely event that there is syntax in the
+        // draft content which our Markdown processor is
+        // unable to process, we delete the draft, so that the
+        // drafts overlay can be opened without any errors.
+        // We also report the exception to the server so that
+        // the bug can be fixed.
+        draft_model.deleteDraft(id);
+        blueslip.error(
+            "Error in rendering draft.",
+            {
+                draft_content: draft.content,
+            },
+            error,
+        );
+        return undefined;
+    }
```

Comment:
The refactor of this function's control flow to do this early seems complex enough that it should be in its own commit.

Reply:
⭐ This is still on my todo list

Reply:
✅